### PR TITLE
Raise an error for zero increment grid

### LINKED ISSUE #1483: Segmentation fault if a grid has zero increments
`grdimage` throwing a segmentation fault insted of some proper error when mistakenly swapping `xarray` y and x coordination.
 

**Full code that generated the error**

```python
X, Y = np.meshgrid(np.arange(10), np.arange(10))
Z = X ** 2 + Y ** 2
da = xr.DataArray(
    data=Z,
    dims=['y', 'x'],
    coords=dict(
        x=(['x'], Y[0, :]),
        y=(['y'], X[:, 0])
    )
)

fig = pygmt.Figure()
fig.basemap(
    frame='a',
    projection=f"X10",
    region=[0, 10, 0, 10]
)
fig.grdimage(da)
fig.show()
```

**Full error message**

```
Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)

```

**System information**

Please paste the output of `python -c "import pygmt; pygmt.show_versions()"`:

```
PyGMT information:
  version: v0.4.1
System information:
  python: 3.7.10 | packaged by conda-forge | (default, Feb 19 2021, 15:59:12)  [Clang 11.0.1 ]
  executable: /Users/yohai/anaconda3/envs/faultSlip/bin/python
  machine: Darwin-18.7.0-x86_64-i386-64bit
Dependency information:
  numpy: 1.19.1
  pandas: 1.1.1
  xarray: 0.18.2
  netCDF4: 1.5.7
  packaging: 21.0
  ghostscript: 9.50
  gmt: 6.2.0
GMT library information:
  binary dir: /Users/yohai/anaconda3/envs/faultSlip/bin
  cores: 12
  grid layout: rows
  library path: /Users/yohai/anaconda3/envs/faultSlip/lib/libgmt.dylib
  padding: 2
  plugin dir: /Users/yohai/anaconda3/envs/faultSlip/lib/gmt/plugins
  share dir: /Users/yohai/anaconda3/envs/faultSlip/share/gmt
  version: 6.2.0
```

### PR DESCRIPTION
**Pull request by yohaimagen**
Raise an error when `DataArray` has zero increment grid.

Fixes #1483 

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by meghanrjones — APPROVED** [review]
Looks good, thanks!

File: pygmt/clib/conversion.py [hunk 1]
```diff
@@ -100,6 +100,10 @@ def dataarray_to_matrix(grid):
                     dim
                 )
             )
+        if coord_inc == 0:
+            raise GMTInvalidInput(
+                "Grid has zero increment in the '{}' dimension.".format(dim)
```
Comment (seisman) [inline pygmt/clib/conversion.py:105]: I think we are slowly converting format strings to f-strings, although there are still many format strings through the code.
```suggestion
                f"Grid has a zero increment in the '{dim}' dimension."
```

**Comment by seisman** [comment]
I changed the PR title to "Raise an error for zero increment grid" so that it better describes the changes in this PR. Feel free to improve the PR title yourself.

File: pygmt/tests/test_clib.py [hunk 2]
```diff
@@ -743,6 +743,23 @@ def test_dataarray_to_matrix_inc_fails():
...
+    """
+    Check that dataarray_to_matrix fails for zero increments grid.
+    """
+    data = np.ones((5, 5), dtype="float64")
+    x = np.linspace(0, 1, 5)
+    y = np.zeros_like(x)
+    grid = xr.DataArray(data, coords=[("y", y), ("x", x)])
+    with pytest.raises(GMTInvalidInput):
+        dataarray_to_matrix(grid)
+    y = np.linspace(0, 1, 5)
```
Comment (weiji14) [inline pygmt/tests/test_clib.py:756]: Add a blank line separating these two sub-tests.
```suggestion
        dataarray_to_matrix(grid)

    y = np.linspace(0, 1, 5)
```

File: pygmt/clib/conversion.py (hunk 1 above)
Comment (yohaimagen) [reply pygmt/clib/conversion.py]: done

**Review by seisman — APPROVED** [review]

**Comment by yohaimagen** [comment]
I think this one is ready to be merged into main, let me know if there is anything else still to be done over here.

**Review by weiji14 — APPROVED** [review]
Thanks @yohaimagen! I'll merge this in now :smiley:

**Pull request merged**